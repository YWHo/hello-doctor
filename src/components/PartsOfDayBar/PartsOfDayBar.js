import React from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';
import {
  DAY_MORNING,
  DAY_AFTERNOON,
  DAY_EVENING
} from '../../shared/constants';
import * as selectors from '../../state/selectors';
import * as actions from '../../state/actions';

const Container = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  height: 38px;
  padding: 0 8px;
  margin: 12px 4px;
`;

const Button = styled.button`
  height: 38px;
  background-color: inherit;
  color: #fff;
  border-color: #fff;
  border-style: ${props => (props.selected ? 'solid' : 'none')};
  border-radius: 50px;
  border-weight: 2px;
  font-size: 18px;
  font-weight: 300;

  :focus {
    outline: none;
  }

  :active {
    outline: none;
  }
`;

export function PartsOfDayBar(props) {
  const { schedules = [], selectedDayPart = DAY_MORNING } = props;

  console.log('schedules: ', schedules);

  return (
    <Container>
      <Button
        selected={selectedDayPart === DAY_MORNING}
        onClick={() => {
          onDayPartClicked(props, DAY_MORNING);
        }}
      >
        Morning
      </Button>
      <Button
        selected={selectedDayPart === DAY_AFTERNOON}
        onClick={() => {
          onDayPartClicked(props, DAY_AFTERNOON);
        }}
      >
        Afternoon
      </Button>
      <Button
        selected={selectedDayPart === DAY_EVENING}
        onClick={() => {
          onDayPartClicked(props, DAY_EVENING);
        }}
      >
        Evening
      </Button>
    </Container>
  );
}

function onDayPartClicked(props, dayPart) {
  const { dSaveSelectedDayPart } = props;
  dSaveSelectedDayPart(dayPart);
}

const mapStateToProps = state => {
  return {
    schedules: selectors.getSchedules(state),
    selectedDayPart: selectors.getSelectedDayPart(state)
  };
};

function mapDispatchToProps(dispatch) {
  return {
    dSaveSelectedDayPart: value => dispatch(actions.saveSelectedDayPart(value))
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(PartsOfDayBar);
