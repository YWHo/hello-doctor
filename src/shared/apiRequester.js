import axios from 'axios';

export function fetchSchedule(date) {
  const fullUrl = `https://frontendchallenge2019.azurewebsites.net/api/Schedule/${date}`;
  return remoteGet(fullUrl);
}

export function fetchProvider(id) {
  const fullUrl = `https://frontendchallenge2019.azurewebsites.net/api/Provider/${id}`;
  return remoteGet(fullUrl);
}

export function fetchPicture(id) {
  const fullUrl = `https://frontendchallenge2019.azurewebsites.net/api/Picture/${id}`;
  return remoteGet(fullUrl);
}

function remoteGet(url) {
  return new Promise((resolve, reject) => {
    axios
      .get(url)
      .then(res => {
        resolve(JSON.parse(res.data));
      })
      .catch(err => reject(err));
  });
}
