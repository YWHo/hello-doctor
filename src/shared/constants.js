export const CLEAR_HAS_AFTERNOON_TIME = 'CLEAR_HAS_AFTERNOON_TIME';
export const CLEAR_HAS_EVENING_TIME = 'CLEAR_HAS_EVENING_TIME';
export const CLEAR_HAS_MORNING_TIME = 'CLEAR_HAS_MORNING_TIME';
export const CLEAR_SCHEDULES = 'CLEAR_SCHEDULES';
export const CLEAR_SELECTED_DATE = 'CLEAR_SELECTED_DATE';
export const CLEAR_SELECTED_DAY_PART = 'CLEAR_SELECTED_DAY_PART';
export const CLEAR_SELECTED_TIME_ID = 'CLEAR_SELECTED_TIME_ID';
export const SAVE_HAS_AFTERNOON_TIME = 'SAVE_HAS_AFTERNOON_TIME';
export const SAVE_HAS_EVENING_TIME = 'SAVE_HAS_EVENING_TIME';
export const SAVE_HAS_MORNING_TIME = 'SAVE_HAS_MORNING_TIME';
export const SAVE_SCHEDULES = 'SAVE_SCHEDULES';
export const SAVE_SELECTED_DATE = 'SAVE_SELECTED_DATE';
export const SAVE_SELECTED_DAY_PART = 'SAVE_SELECTED_DAY_PART';
export const SAVE_SELECTED_TIME_ID = 'SAVE_SELECTED_TIME_ID';
export const TOGGLE_SHOWING_CALENDAR = 'TOGGLE_SHOWING_CALENDAR';
export const DAY_MORNING = 'DAY_MORNING';
export const DAY_AFTERNOON = 'DAY_AFTERNOON';
export const DAY_EVENING = 'DAY_EVENING';
