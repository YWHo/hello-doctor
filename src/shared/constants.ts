export enum ACTION_TYPES {
  CLEAR_PROVIDER_PROFILE = 'CLEAR_PROVIDER_PROFILE',
  CLEAR_SCHEDULES = 'CLEAR_SCHEDULES',
  CLEAR_SELECTED_DATE = 'CLEAR_SELECTED_DATE',
  CLEAR_SELECTED_DAY_PART = 'CLEAR_SELECTED_DAY_PART',
  CLEAR_SELECTED_TIME_ID = 'CLEAR_SELECTED_TIME_ID',
  SAVE_PROVIDER_PROFILE = 'SAVE_PROVIDER_PROFILE',
  SAVE_SCHEDULES = 'SAVE_SCHEDULES',
  SAVE_SELECTED_DATE = 'SAVE_SELECTED_DATE',
  SAVE_SELECTED_DAY_PART = 'SAVE_SELECTED_DAY_PART',
  SAVE_SELECTED_TIME_ID = 'SAVE_SELECTED_TIME_ID',
  TOGGLE_SHOWING_CALENDAR = 'TOGGLE_SHOWING_CALENDAR',
  TOGGLE_SHOWING_PROFILE = 'TOGGLE_SHOWING_PROFILE',
}

export enum DAY {
  AFTERNOON = 'DAY_AFTERNOON',
  EVENING = 'DAY_EVENING',
  MORNING = 'DAY_MORNING',
}
