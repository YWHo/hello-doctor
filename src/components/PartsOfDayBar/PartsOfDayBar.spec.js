import React from 'react';
import { shallow } from 'enzyme';
import { PartsOfDayBar } from './PartsOfDayBar';
import { DAY_AFTERNOON } from '../../shared/constants';

describe('DaySlots', () => {
  it('render correctly', () => {
    const wrapper = shallow(<PartsOfDayBar selectedDayPart={DAY_AFTERNOON} />);
    expect(wrapper).toMatchSnapshot();
  });
});
