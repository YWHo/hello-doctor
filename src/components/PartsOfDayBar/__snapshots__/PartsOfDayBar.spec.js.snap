// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DaySlots render correctly 1`] = `
<styled.div>
  <styled.button
    disabled={true}
    onClick={[Function]}
  >
    Morning
  </styled.button>
  <styled.button
    disabled={true}
    onClick={[Function]}
  >
    Afternoon
  </styled.button>
  <styled.button
    disabled={true}
    onClick={[Function]}
  >
    Evening
  </styled.button>
</styled.div>
`;
